packages/amplify-e2e-tests/src/rds-v2-tests-common/rds-auth-userpool-fields.ts

```diff
@@ -0,0 +1,1925 @@
+      ).rejects.toThrowErrorMatchingInlineSnapshot(expectedOperationError(updateResultSetName, 'Mutation'));
+
+      // non-owner cannot update a record with a dynamic owner list protected field
+      const ownersContentFieldSet = `
+        customId
+        ownersContent
+      `;
+      await expect(
+        async () => await user2TodoHelper.update(updateResultSetName, { ...todo, ownersContent: 'Owners Content' }, ownersContentFieldSet),
+      ).rejects.toThrowErrorMatchingInlineSnapshot(expectedOperationError(updateResultSetName, 'Mutation'));
+
+      // owner cannot read a record with public protected field
+      const getResult1 = await user1TodoHelper.get({ customId: todo['customId'] }, publicFieldSet, false, 'all', 'customId');
+      checkOperationResult(
+        getResult1,
+        { customId: todo['customId'], publicContent: null },
+        `get${modelName}`,
+        false,
+        expectedFieldErrors(['publicContent'], modelName),
+      );
+
+      const listTodosResult1 = await user1TodoHelper.list({}, publicFieldSet, `list${modelName}`, false, 'all');
+      checkListItemExistence(listTodosResult1, `list${modelName}`, todo['customId'], true, 'customId');
+      checkListResponseErrors(listTodosResult1, expectedFieldErrors(['publicContent'], modelName, false));
+
+      /*
+      // non-owner cannot read a record with dynamic owner list protected field in the selection set
+      const ownerReadFieldSet = `
+        customId
+        authors
+        ownersContent
+      `;
+      const getResult2 = await user2TodoHelper.get({ customId: todo['customId'] }, ownerReadFieldSet, false, 'all', 'customId');
+      checkOperationResult(
+        getResult2, 
+        { customId: todo['customId'], authors: null, ownersContent: null }, 
+        `get${modelName}`, 
+        false, 
+        expectedFieldErrors(['authors', 'ownersContent'], modelName)
+      );
+
+      const listTodosResult2 = await user2TodoHelper.list({}, ownerReadFieldSet, `list${modelName}`, false, 'all');
+      checkListItemExistence(listTodosResult2, `list${modelName}`, todo['customId'], true, 'customId');
+      checkListResponseErrors(listTodosResult2, expectedFieldErrors(['authors', 'ownersContent'], modelName, false));
+      */
+
+      // non-owner cannot listen to updates on owner protected fields
+      const todoRandom = {
+        ...todo,
+        customId: Date.now().toString(),
+      };
+      const todoRandomUpdated = {
+        customId: todoRandom.customId,
+        authors: [userName1],
+        privateContent: 'Private Content updated',
+        ownersContent: 'Owners Content updated',
+      };
+      const subscriberClient = getConfiguredAppsyncClientCognitoAuth(graphQlEndpoint, region, userMap[userName2]);
+      const subTodoHelper = createModelOperationHelpers(subscriberClient, schema)[modelName];
+
+      /*
```

Comment:
As above, add an easily searchable todo, or link to this code in our backlog ticket